app/src/main/java/com/fedorrroff/coinpaprikaapp/ui/currencies/GetCoinUseCase.kt

```diff
@@ -0,0 +1,11 @@
+package com.fedorrroff.coinpaprikaapp.ui.currencies
+
+import com.fedorrroff.coinpaprikaapp.models.Coin
+import com.fedorrroff.coinpaprikaapp.repository.Repository
+import io.reactivex.Observable
+import javax.inject.Inject
+
+class GetCoinUseCase @Inject constructor(private val repository: Repository) {
+
+    fun invoke(): Observable<List<Coin>> = repository.getCurrencies()
```

Comment:
используй одинаковый имена, либо Coins либо currencies
```suggestion
    fun invoke(): Observable<List<Coin>> = repository.getCoins()
```